src/typing_extensions.py

```diff
@@ -1381,17 +1336,12 @@ def _clean_optional(obj, hints, globalns=None, localns=None):
                         localns = globalns
                 elif localns is None:
                     localns = globalns
-                if sys.version_info < (3, 9):
-                    original_value = ForwardRef(original_value)
                 else:
                     original_value = ForwardRef(
                         original_value,
                         is_argument=not isinstance(obj, _types.ModuleType)
                     )
```

Comment:
for 1:1 correspondence with the previous logic here, I think this should be

```suggestion

                original_value = ForwardRef(
                    original_value,
                    is_argument=not isinstance(obj, _types.ModuleType)
                )
```

but it's interesting that no tests failed...!